solcjs

```diff
@@ -37,16 +43,19 @@ function abort (msg) {
 }
 
 function readFileCallback(sourcePath) {
-  if (options.basePath)
-    sourcePath = options.basePath + '/' + sourcePath;
-  if (fs.existsSync(sourcePath)) {
-    try {
-      return { 'contents': fs.readFileSync(sourcePath).toString('utf8') }
-    } catch (e) {
-      return { error: 'Error reading ' + sourcePath + ': ' + e };
+  const prefixes = [options.basePath ? options.basePath : ""].concat(options.includePath)
+  for (const prefix of prefixes) {
+    const prefixedSourcePath = (prefix ? prefix + '/' : "") + sourcePath;
+
+    if (fs.existsSync(prefixedSourcePath)) {
+      try {
+        return {'contents': fs.readFileSync(prefixedSourcePath).toString('utf8')}
+      } catch (e) {
+        return {error: 'Error reading ' + prefixedSourcePath + ': ' + e};
+      }
     }
-  } else
-    return { error: 'File not found at ' + sourcePath}
+  }
+  return {error: 'File not found inside the base path or any of the include paths.'}
```

Comment:
Maybe still mention which source could not be found?

Reply:
It's redundant. The part you see here is just a fragment of the message that the callback returns to the compiler. The compiler will prepend a message that already includes the source unit name of the file, which is why I removed it.

I do include it in other messages above because there I know the full path (i.e. source unit name with a base path or an include path prepended) so it actually adds some useful information.